[docs] Update Adding a new module to an existing application section in Expo Modules API: Get started (#28470)
expo/expo: docs/pages/modules/get-started.mdx

```tsx App.tsx
/* @hide ...*/ /* @end */
import { hello } from './modules/my-module';

export default function App() {
  return (
    <View style={styles.container}>
      /* @info In the component, return the result of the native method. */
      <Text>{hello()}</Text>
      /* @end */
    </View>
  );
}
```

Start the development sever in your terminal so that when you edit the native module and build the app in the next step, the changes will be reflected in the app:

<Terminal cmd={['$ npx expo start']} />

Congratulations! You have created a local Expo module. You can now start working on it.

> **info** **Tip**: You can also use absolute import paths [by applying these configuration changes](https://expo.fyi/absolute-path-expo-modules.md).

</Step>

<Step label="3">

### Edit the module

To develop and test your module locally, let's use Android Studio and Xcode for Android and iOS.

#### Android

1. Open the **android** directory (that is generated by `npx expo prebuild` in step 1) from your project in Android Studio. It may take a while for Gradle to finish syncing the native directory project.

> 2. One the project is synced, open **my-module/kotlin+java/expo.modules.mymodule/MyModule** file.

Reviewer: ```suggestion
2. Once the project is synced, open **modules/my-module/android/src/main/java/expo/modules/mymodule/MyModule.kt** file.
```

<Step label="1">

### Create the Expo module

To create a new Expo module from scratch, run the `create-expo-module` script as shown below.
The script will ask you a few questions and then generate the native Expo module along with the example app for Android and iOS that uses your new module.

<Terminal cmd={[`$ npx create-expo-module@latest my-module`]} />

</Step>

<Step label="2">

### Open the module and start the development server

Navigate to the module directory and then open the Android and/or iOS example project by running the following commands:

<Terminal cmd={[`$ cd my-module`, `$ npm run open:android`, `$ npm run open:ios`]} />

Go to **example** directory and start the development sever in your terminal so that when you edit the native module and build the app in the next step, the changes will be reflected in the app:

<Terminal cmd={['$ cd example', '$ npx expo start']} cmdCopy="cd example && npx expo start" />

> **Note:** If you're using Windows, you can open the example project by opening the **android** directory in Android Studio, but you cannot open the iOS project files.

</Step>

<Step label="3">

### Edit the module

#### Android

> 1. Open **my-module/kotlin+java/expo.modules.mymodule/MyModule** file.

Reviewer: ```suggestion
1. Open **my-module/android/src/main/java/expo/modules/mymodule/MyModule.kt** file.
```

### Creating the local Expo module

Navigate to your project directory (the one that contains the **package.json** file) and run the following command. This is the recommended way to create a local Expo module.

<Terminal cmd={[`$ npx create-expo-module@latest --local`]} />

You can provide a meaningful module name in the CLI prompt. For the rest of the prompts, you can also accept the default suggestions.

> **Note**: If you have a pre-existing **ios** directory in your project's root which was created using `npx expo prebuild`, you must reinstall the pods:
>
> <br />

> > <Terminal cmd={[`$ pod install --project-directory=ios`]} />

Author: Replaced here and below to use `npx pod-install`.

Congratulations! You have created a local Expo module. You can now start working on it.

> **Note**: You can also use absolute import paths [by applying these configuration changes](https://expo.fyi/absolute-path-expo-modules.md).

</Step>

<Step label="3">

### Editing the module

To test and develop your module locally, use `npx expo prebuild` to generate the native project directories and use `npx expo run:android` or `npx expo run:ios` to compile and run the example app instead of using EAS Build.

#### Android

If you have an **android** directory in your project that is generated by `npx expo prebuild`, you can open the project directory in Android Studio. Then, you will be able to edit the native code in the **modules/my-module/android/src/main/java/expo/modules/mymodule/** directory.

> For example, change the `hello` method to return a different string such as "Hello world! 🌎🤖". Then, run `npx expo prebuild` to generate the native directory and use `npx expo:android` to run the example app.

Author: ```suggestion
For example, change the `hello` method to return a different string such as "Hello world! 🌎🤖". Then, run `npx expo prebuild` to generate the native directory and use `npx expo:android` to run the example app.
```

> It's not necessary to run `prebuild` after each native change. You only need to `run` it again.

@tsapeta, does the comment mean, we don't need to run `npx expo prebuild` at this step too? (similarly for iOS below).

Reviewer: Yes, there is no need to run `npx expo prebuild`. As we already recommend using Android Studio and Xcode, the easiest workflow in my opinion is to:
- run `npx expo prebuild` if needed (missing `ios` and `android` folders)
- open the project in AS/Xcode
- run `npx expo start` in the example app or your project (local module)
- run a build directly from AS/Xcode
- change native code
- run another build

#### iOS

If you have an **ios** directory in your project that is generated by `npx expo prebuild`, open the project's workspace (**ios/your-app.workspace**) in Xcode. Then, you will be able to edit the native code in **Pods** > **Development Pods** > **MyModule**.

For example, change the `hello` method to return a different string such as "Hello world! 🌎🍎". Then, run `npx expo prebuild` to generate the native directory and use `npx expo:ios` to run the example app.

> Remember you need to either run `npx expo prebuild` each time you make a native change or reinstall the pods using `pod install --project-directory="ios"` (which will be faster).

Reviewer: It's not necessary to run `prebuild` after each native change. You only need to `run` it again.

`pod install` might be necessary when you modify `expo-module.config.json` (e.g. add another module class) and when you add more files to the module (they won't appear in Xcode if you add them using something else than Xcode).
